rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2932,7 +2965,7 @@ extern "C" rmw_ret_t rmw_get_node_names(
   }
 
   std::vector<std::pair<std::string, std::string>> ns;
-  const auto re = std::regex("^name=(.*);namespace=(.*);$", std::regex::extended);
+  const auto re = std::regex("^name=([^;]*);namespace=([^;]*);", std::regex::extended);
```

Comment:
@ivanpauno a hidden bugfix? Maybe we can keep this one in its own commit and rebase-merge afterwards?

Reply:
Not exactly, as it wasn't a bug.
I'm adding `securitycontext=CONTEXT_NAME;` at the end of user data.
The changes are needed because of:
- I want the regex to match `name` and `namespace` also in old versions that doesn't have the `securitycontext` part at the end. So I dropped the `$` from the regex.
- To match correctly with `name=NODE_NAME;namespace=NAMESPACE;securitycontext=CONTEXT;`, I needed to replace the `(.*)` groups with `([^;]*)`. If not, the result would be `name="NODE_NAME"`, `namespace="NAMESPACE;securitycontext=CONTEXT"`.